src/os/filesystem.cc

```diff
@@ -16,75 +16,34 @@
 
 #include "filesystem.hh"
 
+#include <sys/mman.h>
+#include <fcntl.h>
+
 #include <cstdio>
 #include <unistd.h>
+#include <cstdlib>
+#include <cstring>
 #include <iostream>
 #include <filesystem>
 
 namespace os {
-    bool file_exists(const std::string& path) {
-        FILE *file;
-        if ((file = fopen(path.c_str(), "r"))) {
-            fclose(file);
-            return true;
-        }
-        return false;
-    }
-
     std::shared_ptr<char> read_file(const std::string& path) {
-        FILE* file_path = std::fopen(path.c_str(), "r");
-
-        std::fseek(file_path, 0, SEEK_END);
-        int length = std::ftell(file_path);
-        std::fseek(file_path, 0, SEEK_SET);
+        size_t file_size = std::filesystem::file_size(path);
         
-        std::shared_ptr<char> buffer(new char[length + 1], std::default_delete<char[]>());
-        std::fread(buffer.get(), length, 1, file_path);
-        std::fclose(file_path);
-
-        buffer.get()[length] = '\0';
-
-        return buffer;
-    }
+        std::shared_ptr<char> buffer(new char[file_size + 1], std::default_delete<char[]>());
 
-    bool change_dir(const std::string& dir) {
-         return chdir(dir.c_str()) == 0;
-    }
-
-    std::string get_dir() {
-        char* tmp = getcwd(NULL, 0);
-        std::string res(tmp);
-        std::free(tmp);
-        return res;
-    }
+        // Map the file to address space
+        int fd = open(path.c_str(), O_RDONLY);
+        void* mapped_file_addr = mmap(NULL, file_size, PROT_READ, MAP_PRIVATE, fd, 0);
```

Comment:
Is there any way to use mmap through the c++ standard library instead of directly using system calls. Not to be picky, just to maintain portability with other OS.